ppr-ui/src/components/common/PidNumber.vue

```diff
@@ -106,6 +119,31 @@ export default defineComponent({
       localState.pidThree = pidNumberArr[2]
     }
 
+    const dialogRetry = async (retry: boolean): Promise<void> => {
+      localState.showNotFoundDialog = false
+      if (retry) {
+        await validatePid()
+      } else clearPid()
+    }
+
+    const validatePid = async (): Promise<void> => {
```

Comment:
Since it does more than just validation, should probably be validateAndEmitPid()

Reply:
Hmm not sure i agree, we've already pulled out the `emitPid()` into its own functionality. 
If you feel strongly about this, i could return a value and subsequently call `emit()` depending on the result?

Reply:
Yea, that was something I would suggest as well. At first, I thought `validatePid()` would just do one action, return true/false, but then I saw it also calls emit. So hence the comment.

Reply:
Sure, will update.

Reply:
Updated!